src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -50,64 +49,103 @@ public void addNode(T node) {
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, T... tos) {
-    addEdge(from, Arrays.asList(tos));
+  public void addEdge(int weight, T from, T... tos) {
+    addEdge(weight, from, Arrays.asList(tos));
   }
 
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, Iterable<T> tos) {
+  public void addEdge(int weight, T from, Iterable<T> tos) {
```

Comment:
+1 it was my initial intention. But I made it for beeing consistent with the other method: 

```java
public void addEdge(int weight, T from, T... tos)
public void addEdge(int weight, T from, Iterable<T> tos)
```

And:

```java
public void addEdge(T from, T... tos, int weight)
```
is not possible.

Let me know what solution you prefer.